### ISSUE TITLE
Disable client/server check

### ISSUE THREAD

**Issue by jer-sen** [clayne11/eslint-import-resolver-meteor#16]
With nested imports you can now import files in a folder such as "/imports/..." from "/client/..." or "/server/..." within an if block that check Meteor.isServer.
It enables to easily share code between server and client.

This is valid Meteor code and it must not raise error "Unable to resolve path to module ...".

Maybe you can keep this check with non nested imports.
But for nested imports the check must at least be disableable through eslintrc config file.

**Comment by clayne11** [comment]
This is actually still not allowed. I just tested it and importing a `client` file into a `server` file causes a crash. You can see the repro [here](https://github.com/clayne11/meteor-eslint-test).

Closing this issue.

**Comment by jer-sen** [comment]
You should have a look at this : https://guide.meteor.com/1.4-migration.html#nested-imports

They recommand to use things like this :

```
if (Meteor.isClient) {
  import { symbol } from './client-only/file';
}
```

I'm not talking about importing client (or server) only files from server (respectively client)) only files, I'm talking about importing client only or server only files from code shared between client and server (for instance in the /imports directory).

You should reopen this issue.

**Comment by clayne11** [comment]
You actually can't use nested imports with ESLint. Nested imports are against the TC39 spec and the ESLint parser throws a fatal exception trying to parse it. Even if I change the logic it won't work due to the ESLint parser failing.

If you open an issue on ESLint to support nested imports and link it back here I'll reopen this. If/when that issue gets resolved I'll add a flag to support importing client into server files vice versa.

**Comment by jer-sen** [comment]
There is an option (allowImportExportEverywhere) to allow nested imports with ESLint. I'm using it successfully.
You can have a look at the last answer on this page : https://forums.meteor.com/t/weak-dependencies-with-import/25358/7

**Comment by clayne11** [comment]
Oh cool thanks - I had no idea about that flag.

Unfortunately, it doesn't change the fact that importing a client file into a server file causes a crash no matter what. The client files are not bundled into the server bundle by the Meteor build tool.

**Comment by clayne11** [comment]
I updated the reproduction to demonstrate the issue.

**Comment by jer-sen** [comment]
You are probably familiar with this : https://guide.meteor.com/structure.html#javascript-structure
It's true that you can't import in a client (resp. server) only file something from a server (resp. client) only file. A client (resp. server) only file is a file in a folder /client (resp. /server).
But you are allowed to import in a shared file something from a client or server only file. A shared file is a file that is not in a folder /client nor /server.

You should modify your reproduction :
- remove /imports/server and /imports/client folders
- add a file directly in /imports with theses nested imports inside

```
if (Meteor.isClient) {
  import './client/index.js';
}
if (Meteor.isServer) {
  import './server/index.js';
}
```

**Comment by clayne11** [comment]
OK I'll have a look at it when I have some free time, hopefully this weekend.

**Comment by rohithb** [comment]
@clayne11 This client/server checks are causing many eslint errors for perfectly working code.
I'm following this folder structure : https://guide.meteor.com/structure.html#javascript-structure. So, almost all of my code is inside `imports` folder. And pages and components are inside `/import/ui` . In many cases I need to import some client only code (which I defined inside `/imports/helpers/client/`) from `/imports/ui/components`. And as @jay1337 mentioned I'm using importing some server only files ifrom common files using `Meteor.isServer`  like this.
```
if (Meteor.isServer) {
 require('/imports/api/accounts/methods.js');
}
```
All of these are working perfectly but causing `import/no-unresolved` error while linting. 
So it would be great if you provide a way to turn off this rule.

**Comment by clayne11** [comment]
Sorry I haven't gotten to this yet. I will get this done this week.

**Comment by rohithb** [comment]
I partially disabled the server and client checks in pull request #20 .
But its not implemented as a rule, so that this feature can be enabled or disabled as required.

**Comment by clayne11** [comment]
Published as [v0.3.4](https://www.npmjs.com/package/eslint-import-resolver-meteor) thanks to a PR from @rohithb.
